### PR TITLE
Fix promote job quotes and publish tarballs update

### PR DESCRIPTION
**Pull request by ashwindasr**
Publish source tarballs for cli, installer and operator registry

`"""`  and using `${}` allows us to interpolate groovy variables into groovy strings. But if that string contains bash script `$` we need to escape it using `\$`. Thats why the `oc_mirror_extract_cmd` never got an error before, with `"""`

`'''` doesn't let us interpolate groovy variables and keeps it as a raw groovy string.

### CODE REVIEW

**Review by sosiouxme — APPROVED** [review]

**Comment by ashwindasr** [comment]
Tested in [hack space](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ashwin-aos-cd-jobs/job/build%252Fpromote-assembly/7/console)

**Comment by ashwindasr** [comment]
Test [run](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ashwin-aos-cd-jobs/job/build%252Fpromote-assembly/11/console)

**Comment by ashwindasr** [comment]
Directly injected variables to env so as to not confuse groovy and bash interpolation

**Comment by ashwindasr** [comment]
Test [run](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ashwin-aos-cd-jobs/job/build%252Fpromote-assembly/19/console)

**Comment by sosiouxme** [comment]
/lgtm again
we can merge this now, or if you'd like to gain confidence in it we could use the hack space on a release this week to make sure it works first before merging.

**Comment by ashwindasr** [comment]
Yes, that would be great. I'll update my hackspace and will keep back all the code I removed.

**Comment by ashwindasr** [comment]
oc extract wants an empty directory to write to, but the source tarballs were written to the same directory, which caused an issue. Moving that piece of code to after the extract command.

**Pull request merged**
